executor/executor.go

```diff
@@ -343,6 +353,257 @@ func (e *ShowDDLJobsExec) NextChunk(ctx context.Context, chk *chunk.Chunk) error
 	return nil
 }
 
+// RecoverIndexExec represents a recover index executor.
+// It is built from "admin recover index" statement, is used to backfill
+// corrupted index.
+type RecoverIndexExec struct {
+	baseExecutor
+
+	done bool
+
+	tableInfo *model.TableInfo
```

Comment:
Why do we need both `table` and `tableInfo` at the same time? tableInfo = table.Meta()